File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.cpp
class Solution {
public:
    bool uniformArray(vector<int>& nums) {
        int mn = INT_MAX;
        int oddCnt = 0;

        for (int x : nums) {
            if (x < mn)
                mn = x;
            if (x & 1)
                oddCnt++; // check parity using bitwise AND
        }

        // Valid if minimum is odd OR there are no odd numbers
        return (mn & 1) || (oddCnt == 0);
    }
};

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.py
class Solution:
    def uniformArray(self, nums: list[int]) -> bool:
        mn = float("inf")
        odd_cnt = 0

        for x in nums:
            if x < mn:
                mn = x
            if x & 1:
                odd_cnt += 1

        return (mn % 2 == 1) or (odd_cnt == 0)
